ethereum/statetransition/src/test/java/tech/pegasys/teku/statetransition/attestation/MatchingDataAttestationGroupTest.java

```diff
@@ -153,6 +153,54 @@ public void iterator_shouldAggregateAttestationsWithMoreValidatorsFirst() {
             mediumAttestation);
   }
 
+  @Test
+  public void
+      iterator_shouldAggregateAttestationsWithMoreValidatorsFirst_updatingValidatorCountAfterRemoval() {
+    final ValidateableAttestation attestation1 = createAttestation(1);
+    final ValidateableAttestation attestation3 = createAttestation(3);
+
+    final ValidateableAttestation attestationA = addAttestation(1, 3, 5, 7);
+    final ValidateableAttestation attestationB = addAttestation(5, 7, 9);
+    final ValidateableAttestation attestationC = addAttestation(2);
+
+    // Remove some attesters, which should lower attestationA's priority
+    group.remove(attestation1.getAttestation());
+    group.remove(attestation3.getAttestation());
+
+    assertThat(group)
+        .containsExactly(
+            ValidateableAttestation.fromAttestation(
+                aggregateAttestations(
+                    attestationB.getAttestation(), attestationC.getAttestation())),
+            attestationA);
+  }
+
+  @Test
+  public void
+      iterator_shouldAggregateAttestationsWithMoreValidatorsFirst_updatingValidatorCountAfterRemoval_multipleUpdates() {
+    final Attestation toRemove =
+        aggregateAttestations(
+            createAttestation(16).getAttestation(),
+            createAttestation(12).getAttestation(),
+            createAttestation(8).getAttestation(),
+            createAttestation(17).getAttestation(),
+            createAttestation(18).getAttestation(),
+            createAttestation(7).getAttestation());
+
+    final ValidateableAttestation attestationA = addAttestation(19, 0, 4, 8, 12, 16);
+    final ValidateableAttestation attestationB = addAttestation(0, 1, 5, 9, 13, 17);
+    final ValidateableAttestation attestationC = addAttestation(4, 2, 6, 10, 14, 18);
+    final ValidateableAttestation attestationD = addAttestation(0, 3, 7);
+
+    // Remove some attesters, which should reprioritize the attestations
+    group.remove(toRemove);
+
+    final ValidateableAttestation attestationBC =
+        ValidateableAttestation.fromAttestation(
+            aggregateAttestations(attestationB.getAttestation(), attestationC.getAttestation()));
+    assertThat(group).containsExactly(attestationBC, attestationA, attestationD);
```

Comment:
I'm not sure if I understand this test case. The aggregate of attestation B and C seems to be the one with a higher validator count regardless of attestation prioritization. What am I missing?

Reply:
Good catch.  The main point was to just do a more complicated update, but you're right it doesn't make any difference to the final sorting.  Updated.